rfcs/taskforce/0003-Integration-test-infrastructure-spec.md

```diff
@@ -0,0 +1,204 @@
+    3. Logging: Consuming logs from the deployed testnet for validation
+    4. Automated validation: Defines and executes network-wide validations (Out of scope)
+
+The goal is to have a system where each of these components can be independently implemented and the test executive will be responsible for initializing them and facilitating the interactions.
+
+## Design
+
+### Existing
+
+Integrations tests are currently run on a simulated network using local processes. The test controls and inspects the nodes by communicating over RPC. The tests are tightly coupled to the simulated network and cannot be ported to run on cloud-based or container-based environments. More information on the motivation is described in this [rfc](https://github.com/CodaProtocol/coda/pull/4715) (TODO: Copy it to make it easier to read?)
+
+### Improvements
+
+The proposal for the test executive is part of the whole integration test framework revamp to address the drawbacks of the existing framework (#4735).
+
+### Proposal
+
+#### Infrastructure configuration
+
+#### Test executive
+
+The test executive requires the following inputs to successfully execute an integration test
+
+1. Genesis constants and ledger:
+`Runtime_config.t` currently has all the configurable constants and ledger
+
+A ledger can be specified in any of the following way: (already implemented)
+
+```ocaml
+
+module Accounts = struct
+  type single =
+    { pk: string option [@default None]
+    ; sk: string option [@default None]
+    ; balance: Currency.Balance.t
+    ; delegate: string option [@default None] }
+end
+
+module Ledger = struct
+  type base =
+    | Named of string  (** One of the named ledgers in [Genesis_ledger] *)
+    | Accounts of Accounts.t  (** A ledger generated from the given accounts *)
+    | Hash of string  (** The ledger with the given root hash *)
+end
+```
+
+Therefore, one could either use compiled ledgers or generate a new json list using the Testnet SDK from coda_automation.
+
+2.Network config: the configuration for the network the integration test will run on.
+
+```ocaml
+
+(*flags that we may want to specify for each daemon*)
+
+type block_producer_flags = {
+      coinbase_receiver: Public_key.Compressed.t
+}
+
+type snark_coordinator_flags = {
+      snark_worker_fee: Fee.t
+    ; work_reassignment_wait: int
+    ; work_selection: `Seq | `Random
+}
+(* The block producer keys would be 1 to [count] in the ledger and any non-default flags*)
+type block_producers = {count : int; flags: block_producer_flags list }
+
+type archive_nodes = {count: int}
+
+(*Snark coordinator keys will be from [block_producers.count]+1 to [block_producers.count]+[count]. Each snark coordinator is associated with a key and a fee. All the proofs generated by the workers connected to the coordinator is associated with that key and the fee. Each coordinator has a set of workers that would get work from the coordinator and submit proofs for the same.*)
```

Comment:
Might it be worth using explicit indices or keys for these? For example, it might be useful to test behaviour for snark coordinators without accounts in the ledger, or for nodes with a snark coordinator and block producer associated with the same public key.